### PR TITLE
netease-music-tui: init at v0.1.2

### PR DESCRIPTION
**Pull request by Vonfry**
###### Motivation for this change
New package.

###### Things done

- [X] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [X] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Details
***closure size***
```
/nix/store/agrz6qbrmc3fabrw9s2f3ihjzs6l7ncw-netease-music-tui-v0.1.2	   49127512
```

***nixpkgs-review***
```
1 package added:
netease-music-tui-v0.1.2

1 package built:
netease-music-tui
```

### CODE REVIEW

File: pkgs/applications/audio/netease-music-tui/default.nix [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  version = "v0.1.2";
+
+  src = fetchFromGitHub {
+    owner = "betta-cyber";
+    repo = "netease-music-tui";
+    rev = "${version}";
+    sha256 = "0m5b3q493d32kxznm4apn56216l07b1c49km236i03mpfvdw7m1f";
+  };
+
+  cargoPatches = [ ./cargo-lock.patch ];
```
Comment (SuperSandro2000) [inline pkgs/applications/audio/netease-music-tui/default.nix:14]: Did you submit this as a PR? If so fetch that patch.

File: pkgs/applications/audio/netease-music-tui/default.nix [hunk 2]
```diff
@@ -0,0 +1,29 @@
+{ fetchFromGitHub, rustPlatform, lib, alsaLib, pkg-config, openssl }:
+
+rustPlatform.buildRustPackage rec {
+  name = "netease-music-tui-${version}";
```
Comment (SuperSandro2000) [inline pkgs/applications/audio/netease-music-tui/default.nix:4]:
```suggestion
  pname = "netease-music-tui";
```

File: pkgs/applications/audio/netease-music-tui/default.nix [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+  cargoSha256 = "1kfbnwy3lkbhz0ggxwr5n6qd1plipkr1ycr3z2r7c0amrzzbkc7l";
+  verifyCargoDeps = true;
+
+  meta = with lib; {
+    homepage = "https://github.com/betta-cyber/netease-music-tui";
+    description = "netease cloud music terminal client by rust";
+    maintainers = with maintainers; [ vonfry ];
+    license = licenses.mit;
+    platforms = with platforms; flatten [ linux darwin ];
```
Comment (SuperSandro2000) [inline pkgs/applications/audio/netease-music-tui/default.nix:27]:
```suggestion
```
This is the same as buildRustPackage sets.

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 107560` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>netease-music-tui</li>
  </ul>
</details>

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 107560` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package marked as broken and skipped:</summary>
  <ul>
    <li>netease-music-tui</li>
  </ul>
</details>

File: pkgs/applications/audio/netease-music-tui/default.nix (hunk 1 above)
Comment (Vonfry) [reply pkgs/applications/audio/netease-music-tui/default.nix:14]: The pr in upstream is on readme file about prerequisites. The patch that used in this pr is on cargo.lock, which is similar to other packages by rust without a lock file in original repo.

**Comment by Vonfry** [comment]
> Result of `nixpkgs-review pr 107560` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
> 
> 1 package marked as broken and skipped:
> * netease-music-tui

Should I remove darwin from platforms? The package supports it, but I haven't device to test it.

File: pkgs/applications/audio/netease-music-tui/default.nix (hunk 1 above)
Comment (SuperSandro2000) [reply pkgs/applications/audio/netease-music-tui/default.nix:14]: Can you submit it so that we can fetch it?

**Comment by SuperSandro2000** [comment]
> > Result of `nixpkgs-review pr 107560` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
> > 1 package marked as broken and skipped:
> > 
> > * netease-music-tui
> 
> Should I remove darwin from platforms? The package supports it, but I haven't device to test it.

This simply means it can't be build on darwin because I think alsa is not available on there.

File: pkgs/applications/audio/netease-music-tui/default.nix (hunk 1 above)
Comment (Vonfry) [reply pkgs/applications/audio/netease-music-tui/default.nix:14]: The author ignore that file from repo. And take a look like [othter rust package on nixpkgs](https://github.com/NixOS/nixpkgs/blob/2c931312ce6c7baf91ec98e164bef0241fc20aa6/pkgs/applications/misc/effitask/default.nix), I do the same thing. Why must we fetch it from different place.

Comment (SuperSandro2000) [reply pkgs/applications/audio/netease-music-tui/default.nix:14]: I wanted to upstream the change to not necessarily add the large patch to the git but it is no required.

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 107560` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>netease-music-tui</li>
  </ul>
</details>

**Pull request merged**